packages/datadog-plugin-aws-sdk/test/index.spec.js

```diff
@@ -0,0 +1,1084 @@
+'use strict'
+
+const agent = require('../../dd-trace/test/plugins/agent')
+const plugin = require('../src')
+const fixtures = require('./fixtures/base')
+const semver = require('semver')
+
+wrapIt()
+
+const sort = spans => spans.sort((a, b) => a.start.toString() >= b.start.toString() ? 1 : -1)
+
+const closeAndWipeAgent = () => {
+  agent.close()
+  agent.wipe()
+}
+
+const baseSpecCheck = (done, agent, operation, serviceName, klass, fixture, key, metadata, config) => {
```

Comment:
I'd say to move these to an external file since they make it difficult to see where the tests actually begin. Also, it's possible to move the entire `it` in the function as well to make the tests even cleaner.

Reply:
It may even be possible to have a function that wraps all the others if every test uses the same ones.

Reply:
i've extracted these into a separate spec_helper file. However, basically anytime I extract anything at the size of an `it` block or larger into it's own re-usable function mocha executes things in the wrong order, so I think I would lean toward separating each service into it's own spec file if we feel the size here still makes this too large to reason about.

Reply:
Actually I think I was able to abstract this out a bit more, just had some scope issues that I had missed. Will push up an update here shortly

Reply:
Pushed up some nice little abstractions around the super repetitive chunks of tests for each service, should be  much easier to reason about now and extend to new services going forward

Reply:
Did you solve the issue with `it` ordering? The way it works, it will always execute in the correct order unless `it` was not called in the right place, which shouldn't happen given the code is synchronous.

Reply:
Yes, the issue was that the `before` `after` hooks were behaving strangely when I removed the `it` blocks into their own helper function, but I was able to resolve the issue and things are now executing in the appropriate order.